crates/api_models/src/payments.rs

```diff
@@ -7841,3 +7859,36 @@ mod billing_from_payment_method_data {
         assert!(billing_details.is_none());
     }
 }
+
+#[cfg(feature = "v2")]
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize, ToSchema)]
+pub struct PaymentIntentRevenueRecoveryMetadata {
+    ///Total number of billing connector + recovery retries for a payment intent.
+    #[schema(value_type = i32,example = "1")]
+    pub retry_count: i32,
```

Comment:
Nit: I guess this can be a `u16` as well, I don't think we'll have more than 65,535 retries. Same change can be done on the diesel model side as well.